### DISCUSSION TITLE
tmp directory in tasks

### DISCUSSION THREAD

**Discussion by JulesVandenbroeck** [columnflow/columnflow#580] [Q&A]
Hi everyone,

I had a quick question regarding temporary files that are used in tasks to save intermediate files before saving the final ouput of the task. It seems that these files are not automatically deleted even after the task has finished. Is there a specific reason these files are kept after the task are finished or is it possible to configure this so they are deleted right after a task finishes. From looking at the `cf.ProduceColumns` task for example, the following lines initiate the tmp file:
```
tmp_dir = law.LocalDirectoryTarget(is_tmp=True)
tmp_dir.touch()
```
However the tmp file is not explicitly deleted after writing to the output. Is this done outside the run function of the Class or not at all? This same two lines are also used repetitively in other tasks even though no tmp file is used in writing out data. 

Update:

As this seems to be an issue on our (Ghent) side. This is the configuration of the tmp directory in `law.cfg`:

```
[target]

tmp_dir: /scratch/$CF_CERN_USER
tmp_dir_perm: 755
```

Kind Regards,
Jules Vandenbroeck

**Comment by pkausw** [comment]
Hi @JulesVandenbroeck ,

afaik, `LocalDirectoryTargets` that are initialized with `is_tmp=True` are automatically deleted by law as soon as the object goes out of scope. Maybe @riga can comment here.

It's a good question why it seems like this mechanism doesn't work for you. Are none of the temporary outputs deleted at all?

**Reply by JulesVandenbroeck** [reply]
None of the temporary outputs are deleted for me or other colleagues at Ghent. This is the reason for raising the discussion here as the tmp directory is getting flooded with temporary outputs slowing the pipeline down when a task needs to access this directory.
